Resources/doc/messenger.rst

```diff
@@ -0,0 +1,11 @@
+Messenger integration
+=====================
+
+The bundle automatically registers a `Messenger`_ middleware that clears
+document managers between messages being handled to avoid outdated data.
```

Comment:
Suggestion:

> ...that clears all document managers after handling messages, which helps to isolate each handler and guard against reading out-of-date document data.

I summarized that from @weaverryan's explanation in https://github.com/symfony/symfony/pull/34156. You could also considering adding a link to the issue for more context, since there isn't much else to say on this page, but I don't feel strongly either way about that.

Reply:
Dang, now I see it's not even a middleware but rather a event subscriber 👀 I'll go with your summary

Reply:
Looking at the Symfony PR, I think the design changed somewhere along the way. In any event, current text LGTM.